pkg/config/util.go

```diff
@@ -0,0 +1,117 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/rs/zerolog/log"
+)
+
+type unmarshalContext interface {
+	IsSet(string) bool
+	Get(string) interface{}
+	AllKeys() []string
+}
+
+// unmarshalCompositeKey takes a key and an output structure to unmarshal into. It gets the
+// composite value associated with the given key and decodes it into the provided output structure.
+// It's especially useful when the desired value is not directly associated with the key, but
+// instead is spread across various nested sub-keys within the configuration.
+func unmarshalCompositeKey(v unmarshalContext, key string, output interface{}) error {
```

Comment:
not sure I understand this?

Reply:
I am only moving this code to a different file. This isn't anything new in the PR.